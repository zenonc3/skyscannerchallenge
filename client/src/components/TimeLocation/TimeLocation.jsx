import React from "react";
import BpkText from "bpk-component-text";
import STYLES from "./TimeLocation.scss";

const c = className => STYLES[className] || 'UNKNOWN';

export default class TimeLocation extends React.Component {
    render() {
        return (
            <div className={c("TimeLocation_container")}>
                <div>
                    <BpkText className={c("TimeLocation_time")}> 23:59 </BpkText>
                </div>
                <div>
                    <BpkText className={c("TimeLocation_location")} bold={true}> LOC </BpkText>
                </div>
            </div>
        );
    }
}