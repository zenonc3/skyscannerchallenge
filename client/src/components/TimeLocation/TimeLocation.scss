@import '~bpk-mixins';

.TimeLocation {
  &__time {
    width: 25pt;
    height: 25pt;
  }

  &__location {
    width: 25pt;
    height: 25pt;
    color: grayscale($color: #777);
  }

  &__container {
    padding-left: 2pt;
  }
}
