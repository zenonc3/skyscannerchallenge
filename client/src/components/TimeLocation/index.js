import TimeLocation from './TimeLocation';

export default TimeLocation;
