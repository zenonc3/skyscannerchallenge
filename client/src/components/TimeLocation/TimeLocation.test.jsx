import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import TimeLocation from './TimeLocation';

describe('TimeLocation', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TimeLocation time="2019-10-21T12:55:00" location="EDI" />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

