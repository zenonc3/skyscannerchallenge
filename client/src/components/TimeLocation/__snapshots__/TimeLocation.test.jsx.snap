// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLocation should render correctly 1`] = `
<div
  className="TimeLocation__container"
>
  <div>
    <BpkText
      bold={false}
      className="TimeLocation__time"
      tagName="span"
      textStyle="base"
    >
       
      12:55
       
    </BpkText>
  </div>
  <div>
    <BpkText
      bold={true}
      className="TimeLocation__location"
      tagName="span"
      textStyle="base"
    >
       
      EDI
       
    </BpkText>
  </div>
</div>
`;
